### ISSUE TITLE
Failed to extract mpk in Death Stranding

### ISSUE THREAD

**Issue by hinwai** [Jayveer/Decima-Explorer#30]
tried to list the files first using:
`DecimaExplorer.exe -list movies.mpk`

But output
`Error: Failed to find initial archive`
`File table extracted to file_list.txt`

And the txt file is 0byte.

**Comment by Jayveer** [comment]
Currently there isn't a list for movies as I'm not sure of what the hashed names are, therefore its only possible to extract movies via the ID, i.e if we wanted to extract movie at index 0;

```
DecimaExplorer.exe -extract input.mpk 0 output.bk2
```

I think there are a total of 40 movies.

**Comment by kotori2** [comment]
smh not working for me
```
Movies> ./DecimaExplorer.exe -extract movies.mpk 0 output.bk2
Running Decima Explorer v2.7:
Visit https://github.com/Jayveer/Decima-Explorer for updates:

Error: Input file is of an unrecognized format
```
This file have an header of `00 7C 13 51` and not maching other files.

**Comment by Jayveer** [comment]
I guess something broke with an update. You can use version 2.1 for now if you want to extract movies.

**Comment by nayred** [comment]
Same problem, it doesn't work for me either. Nor version 2.7. nor version 2.1. Just nothing happens. I'm desperate. If someone succeeds in unpacking, be so kind as to upload movie archive to the some filehosting.

**Comment by w2jmoe** [comment]
> 
> 
> Currently there isn't a list for movies as I'm not sure of what the hashed names are, therefore its only possible to extract movies via the ID, i.e if we wanted to extract movie at index 0;
> 
> ```
> DecimaExplorer.exe -extract input.mpk 0 output.bk2
> ```
> 
> I think there are a total of 40 movies.



> 
> 
> Currently there isn't a list for movies as I'm not sure of what the hashed names are, therefore its only possible to extract movies via the ID, i.e if we wanted to extract movie at index 0;
> 
> ```
> DecimaExplorer.exe -extract input.mpk 0 output.bk2
> ```
> 
> I think there are a total of 40 movies.

Yes, I extracted all the game videos. There are 40 .bk2 files and RADVideo for playing .bk2 videos.
Thank you very much!! @Jayveer

https://pic.downk.cc/item/5f9849881cd1bbb86b3ec382.png
https://pic.downk.cc/item/5f9849881cd1bbb86b3ec37e.png

**Comment by nayred** [comment]
> Yes, I extracted all the game videos. There are 40 .bk2 files and RADVideo for playing .bk2 videos.
> Thank you very much!! @Jayveer
> 
> https://pic.downk.cc/item/5f9849881cd1bbb86b3ec382.png
> https://pic.downk.cc/item/5f9849881cd1bbb86b3ec37e.png

It helps. Thanks for the instructions!

**Comment by stevebow** [comment]
> Yes, I extracted all the game videos. There are 40 .bk2 files and RADVideo for playing .bk2 videos. Thank you very much!! @Jayveer
> 
> https://pic.downk.cc/item/5f9849881cd1bbb86b3ec382.png
> https://pic.downk.cc/item/5f9849881cd1bbb86b3ec37e.png

Which version of sg-unpack did you use?  I tried V1.0.0.0 from [https://github.com/rdavisau/sg-unpack/releases](url) , but it fails:

`E:\Games\Steam Games SSD\steamapps\common\DEATH STRANDING DIRECTORS CUT\Movies>sg-unpack.exe -i movies.mpk`
`Using inputs [ movies.mpk ] in .`
`--output-path not specified. Listing contents only.`
` `
`Scanning 1 inputs...`
` `
`Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.Exception: An error occurred while attempting to check MPAK header of 'movies.mpk'`
`   at SgUnpack.MPakReader.Process(String f)`
`   at System.Linq.Parallel.SelectQueryOperator 2.SelectQueryOperatorResults.GetElement(Int32 index)`
`   at System.Linq.Parallel.ArrayMergeHelper 1.ToArrayElement(Int32 index)`
`[...]`

I am trying to extract movies.mpk from the PC version of DS Director's cut.  Did you use the same or was it a console version, or perhaps DS original?

**Comment by EwilCZ** [comment]
> > Yes, I extracted all the game videos. There are 40 .bk2 files and RADVideo for playing .bk2 videos. Thank you very much!! @Jayveer
> > https://pic.downk.cc/item/5f9849881cd1bbb86b3ec382.png
> > https://pic.downk.cc/item/5f9849881cd1bbb86b3ec37e.png
> 
> Which version of sg-unpack did you use? I tried V1.0.0.0 from [https://github.com/rdavisau/sg-unpack/releases](url) , but it fails:
> 
> `E:\Games\Steam Games SSD\steamapps\common\DEATH STRANDING DIRECTORS CUT\Movies>sg-unpack.exe -i movies.mpk` `Using inputs [ movies.mpk ] in .` `--output-path not specified. Listing contents only.` ` ` `Scanning 1 inputs...` ` ` `Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.Exception: An error occurred while attempting to check MPAK header of 'movies.mpk'` ` at SgUnpack.MPakReader.Process(String f)` ` at System.Linq.Parallel.SelectQueryOperator 2.SelectQueryOperatorResults.GetElement(Int32 index)` ` at System.Linq.Parallel.ArrayMergeHelper 1.ToArrayElement(Int32 index)` `[...]`
> 
> I am trying to extract movies.mpk from the PC version of DS Director's cut. Did you use the same or was it a console version, or perhaps DS original?

You don't need to use sg-unpack. Just use Decima Explorer 2.1 and then these commands:

DecimaExplorer.exe -extract movies.mpk 0 0.bk2
DecimaExplorer.exe -extract movies.mpk 1 1.bk2
DecimaExplorer.exe -extract movies.mpk 2 2.bk2

Etc.

Worked for me :)

**Comment by stevebow** [comment]
> You don't need to use sg-unpack. Just use Decima Explorer 2.1 and then these commands:
> 
> DecimaExplorer.exe -extract movies.mpk 0 0.bk2 DecimaExplorer.exe -extract movies.mpk 1 1.bk2 DecimaExplorer.exe -extract movies.mpk 2 2.bk2
> 
> Etc.
> 
> Worked for me :)

Yes, that's what I ended up doing too in the end, thanks.  I found that DecimaExplorer 2.7 won't extract the movies, but 2.1 did, so something broke somewhere between 2.1 and 2.7.

**Comment by RaduNastase** [comment]
Hey guys, there are actually 41 vids, not 40.

I extracted all of them and recoded at 30 fps instead of 60 (this fixes the annoying bug where, if you want to play at 30 fps, these fmvs would run at 15 fps instead of 30), renamed them as this gentleman here did  https://www.nexusmods.com/deathstranding/mods/87?tab=files

..and they work well, except i have no clue how to integrate the missing file (no. 31 as extracted, the scene is at the very end of the game, after credits i think).

Any ideea how to do it?
